assets/javascripts/discourse/pre-initializers/rich-editor.extension.js

```diff
@@ -0,0 +1,140 @@
+import { isTesting } from "discourse/lib/environment";
+import { withPluginApi } from "discourse/lib/plugin-api";
+import I18n, { i18n } from "discourse-i18n";
+import guessDateFormat from "discourse/plugins/discourse-calendar/discourse/lib/guess-best-date-format";
+
+const EVENT_ATTRIBUTES = {
+  name: { default: null },
+  start: { default: null },
+  end: { default: null },
+  reminders: { default: null },
+  minimal: { default: null },
+  closed: { default: null },
+  status: { default: "public" },
+  timezone: { default: "UTC" },
+  allowedGroups: { default: null },
+};
+
+/** @type {RichEditorExtension} */
+const extension = {
+  nodeSpec: {
+    event: {
+      attrs: EVENT_ATTRIBUTES,
+      group: "block",
+      defining: true,
+      isolating: true,
+      draggable: true,
+      parseDOM: [
+        {
+          tag: "div.discourse-post-event",
+          getAttrs(dom) {
+            return { ...dom.dataset };
+          },
+        },
+      ],
+      toDOM(node) {
+        const dataAttrs = Object.entries(node.attrs).reduce(
+          (acc, [key, value]) => {
+            if (value !== null) {
+              acc[`data-${key.replace(/([A-Z])/g, "-$1").toLowerCase()}`] =
```

Comment:
I think we can avoid this camel<>kebab conversion if we use a different strategy, but creating the element itself instead of returning the array notation.

Something like...

```javascript
const element = document.createElement("div");
for (const [key, value] of Object.entries(node.attrs)) {
  if (value !== null) {
    element.dataset[key] = value;
  }
}
```

Then you can `element.appendChild` and `return element` in the end.